internal/terraform/module/module_manager.go

```diff
@@ -97,16 +99,38 @@ func (mm *moduleManager) EnqueueModuleOp(modPath string, opType op.OpType) error
 	return nil
 }
 
-func (mm *moduleManager) SchemaForModule(modPath string) (*schema.BodySchema, error) {
+func (mm *moduleManager) SchemaForModule(modPath string, languageID string) (*schema.BodySchema, error) {
 	mod, err := mm.ModuleByPath(modPath)
 	if err != nil {
 		return nil, err
 	}
 
-	return schemaForModule(mod, mm.schemaStore)
+	if languageID == "tfvars" {
+		return schemaForTfVars(mod)
+	} else {
+		return schemaForTerraform(mod, mm.schemaStore)
+	}
+}
+
+func schemaForTfVars(mod *state.Module) (*schema.BodySchema, error) {
```

Comment:
The implementation is AOK, but I think this could live somewhere here https://github.com/hashicorp/terraform-schema/tree/main/schema (e.g. `variables_schema.go`); exported as `SchemaForVariables(vars map[string]tfmod.Variable) (*schema.BodySchema, error)`

Then the implementation here would be practically just a single line, leaving the important bits (about how the schema is assembled) in `terraform-schema`.

--- 

Side note: I reckon we would benefit from adding `IsSensitive` to the [`AttributeSchema`](https://pkg.go.dev/github.com/hashicorp/hcl-lang@v0.0.0-20210525083843-b62f3a996576/schema#AttributeSchema), which could then be also reflected in completion and hover somehow - e.g. displaying it next to the type and other flags (optional/required/computed)
![Screenshot 2021-06-01 at 13 07 46](https://user-images.githubusercontent.com/287584/120320757-70705080-c2da-11eb-80fd-1fe488fe42e9.png)
![Screenshot 2021-06-01 at 13 08 06](https://user-images.githubusercontent.com/287584/120320763-7108e700-c2da-11eb-81d6-be7c8a60629c.png)

That would also help us flag up the provider-flagged sensitive attributes here https://github.com/hashicorp/terraform-schema/blob/18f435878cf26e12bff5dcec121b030518997422/schema/convert_json.go#L104

But feel free to leave this particular change in `hcl-lang` for later - I'm just thinking out loud here! 😄

Reply:
Thanks for the review! :) PR to add IsSensitive to attribute schema: https://github.com/hashicorp/hcl-lang/pull/42

Reply:
PR to implement `SchemaForVariables()` in terraform-schema: https://github.com/hashicorp/terraform-schema/pull/49 (will add `isSensitive` when the one above is merged).